### PR TITLE
fix(environments): Fix release stats

### PR DESCRIPTION
**Pull request by lynnagara**
Ensure release stats component is rerendered when environment changed
Set first seen and last seen data correctly

### CODE REVIEW

File: src/sentry/static/sentry/app/components/group/releaseStats.jsx [hunk 1]
```diff
@@ -67,7 +67,8 @@ const GroupReleaseStats = createReactClass({
       this.state.loading !== nextState.loading ||
       this.state.error !== nextState.error ||
       this.state.environment !== nextState.environment ||
-      this.props.group.id !== nextProps.group.id
+      this.props.group.id !== nextProps.group.i ||
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/group/releaseStats.jsx]: group.i < id

is this the entire state and props? what is it here that we don't want to update one? could we delete the method or use a deep compare util?

**Review by MaxBittker — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/components/group/releaseStats.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/components/group/releaseStats.jsx]: Agh, thanks

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/group/releaseStats.jsx]: I'm not sure why we need this. It seems that we're just comparing all the props and state that would reasonably change anyway.

**Review by MaxBittker — APPROVED** [review]
approving this because it fixes the bug and doesn't make it worse, but it would be nice to figure out why we need componentShouldUpdate at all since we might forget one of these in the future

**Pull request merged**